### PR TITLE
Improve store management and add support for store specific settings

### PR DESCRIPTION
**Pull request by vogti**

### CODE REVIEW

File: csv-adapter/src/main/java/ch/unibas/dmi/dbis/polyphenydb/adapter/csv/CsvStore.java [hunk 1]
```diff
@@ -103,4 +111,41 @@ public void updateColumnType( Context context, CatalogColumn catalogColumn ) {
...
+                        settings.put( s.name, newValue );
+                    } else {
+                        // Same value, do nothing
+                    }
+                } else {
+                    throw new RuntimeException( "Setting \"" + s.name + "\" is not modifiable!" );
+                }
+            }
+        }
+    }
```
Comment (schoenja) [inline csv-adapter/src/main/java/ch/unibas/dmi/dbis/polyphenydb/adapter/csv/CsvStore.java]: A lot of this seems to be code that should be extracted into a more generic form. Afaict the only part that will differ between stores is the `switch` statement in the middle (L133 - L140).

So we could probably move the outer part into abstract `Store` class, and then have a private abstract function `applySetting`?

Comment (vogti) [reply csv-adapter/src/main/java/ch/unibas/dmi/dbis/polyphenydb/adapter/csv/CsvStore.java]: You're absolutely right. This is definitely not a nice solution. Thanks for pointing out. It is now as you suggested.

File: core/src/main/java/ch/unibas/dmi/dbis/polyphenydb/Store.java [hunk 2]
```diff
@@ -47,4 +53,94 @@ public Store( final int storeId, final String uniqueName ) {
...
+
+    public void updateSettings( Map<String, String> newSettings ) {
+        for ( AdapterSetting s : getAvailableSettings() ) {
+            if ( newSettings.containsKey( s.name ) ) {
+                if ( s.modifiable ) {
+                    String newValue = newSettings.get( s.name );
+                    if ( !s.canBeNull && newValue == null ) {
+                        throw new RuntimeException( "Setting \"" + s.name + "\" is not allowed to be null!" );
+                    }
+                    if ( newValue.equals( settings.get( s.name ) ) ) {
```
Comment (schoenja) [inline core/src/main/java/ch/unibas/dmi/dbis/polyphenydb/Store.java]: Shouldn't this be negated?

```suggestion
                    if ( ! newValue.equals( settings.get( s.name ) ) ) {
```

Comment (vogti) [reply core/src/main/java/ch/unibas/dmi/dbis/polyphenydb/Store.java]: Oh yes... Thx

**Pull request merged**